src/radar_control.py

```diff
@@ -681,6 +718,76 @@ def run_cfs_scan(
     return freq_data
 
 
+def cfs_block(ave_params, cfs_params_dict, cfs_sockets):
+    aveperiod = ave_params.aveperiod
+    if (
+        cfs_params_dict[aveperiod].last_cfs_set_time[aveperiod.beam_iter]
+        < time.time() - ave_params.aveperiod.cfs_stable_time
+        or aveperiod.cfs_always_run
+    ):
+        # Only let CFS run after the user set stable time has
+        # passed to prevent CFS from switching freqs to quickly
+        ave_params.sequence = aveperiod.cfs_sequence
+        ave_params.decimation_scheme = aveperiod.cfs_sequence.decimation_scheme
+
+        freq_spectrum = run_cfs_scan(
+            cfs_sockets[0],
+            cfs_sockets[1],
+            ave_params,
+            cfs_sockets[2],
+        )
+        ave_params.num_sequences += 1
```

Comment:
Yes it needs to be incremented. Due to the condition we currently use to trigger cfs (ie `aveperiod.num_sequences == 0`) if we do not increment this than all the radar does is scan and never actually gets past the cfs block. Additionally, the sequence number is very closely tied to handling of which sequences to write in data write. Currently in data write we tell it to skip the sequence numbers that correspond to which sequence ran the cfs scan. 

There might be a way to untangle these, but I'm not sure how deep one would have to go to do so. I'll look into it a little bit today, but it might be a better task for another branch more dedicated to it.

Reply:
Oh right I remember this now, sorry for bringing it up. So what will the value `num_sequences` be in the data file if, for example, the averaging period ran for 37 sequences, not including an initial CFS sequence? 37 or 38?